contractcourt/chain_watcher.go

```diff
@@ -488,6 +489,19 @@ func (c *chainWatcher) toSelfAmount(tx *wire.MsgTx) btcutil.Amount {
 	return selfAmt
 }
 
+// chanSyncMsg fetches the latest channel sync message for the channel.
+func (c *chainWatcher) chanSyncMsg() (*lnwire.ChannelReestablish, error) {
+	lnChannel, err := lnwallet.NewLightningChannel(
+		nil, nil, c.cfg.chanState,
+	)
+	if err != nil {
+		return nil, err
+	}
+	defer lnChannel.Stop()
```

Comment:
Bit of a code smell that we need to do this (make a set of goroutines) just to grab this field. Instead we can [efactor this method slightly](https://github.com/lightningnetwork/lnd/blob/5803da382a4d16c155cd5396d585c0340736ecad/lnwallet/channel.go#L3449:29) to make a (possible private version) that accepts the channel state and commitments heights and use that in place. Recall the issues we had in the funding manager that caused us to just pass around the channel state everywhere vs the full state machine.

Reply:
Fixed!